libs/execution_context/ebpf_program.c

```diff
@@ -1431,7 +1450,12 @@ ebpf_program_invoke(
 {
     if (ebpf_program_disable_invoke) {
         *result = 0;
-        return;
+        return EBPF_EXTENSION_FAILED_TO_LOAD;
+    }
+
+    if (ReadPointerNoFence((void* const volatile*)(&program->extension_program_data)) == NULL) {
```

Comment:
Perhaps this is obvious to maintainers, but would a comment clarifying how this check interacts with the epoch protection be helpful?

I don't quite follow how the `ebpf_epoch_synchronize` in `_ebpf_program_type_specific_program_information_detach_provider` protects the program invocation thread when the detach routine could clear the pointer and free its memory at any time during an epoch.

Reply:
Hope this helps.